controllers/osbuildenvconfig_controller.go

```diff
@@ -18,24 +18,109 @@ package controllers
+	certmanagerv1 "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
+	certmanagermetav1 "github.com/cert-manager/cert-manager/pkg/apis/meta/v1"
+	"github.com/go-logr/logr"
+
+	osbuildv1alpha1 "github.com/project-flotta/osbuild-operator/api/v1alpha1"
+	"github.com/project-flotta/osbuild-operator/internal/conf"
+	"github.com/project-flotta/osbuild-operator/internal/repository/certificate"
+	"github.com/project-flotta/osbuild-operator/internal/repository/configmap"
+	"github.com/project-flotta/osbuild-operator/internal/repository/deployment"
+	"github.com/project-flotta/osbuild-operator/internal/repository/osbuildenvconfig"
+	"github.com/project-flotta/osbuild-operator/internal/repository/secret"
+	"github.com/project-flotta/osbuild-operator/internal/repository/service"
+	"github.com/project-flotta/osbuild-operator/internal/templates"
+)
+
+const (
+	osBuildOperatorFinalizer = "osbuilder.project-flotta.io/osBuildOperatorFinalizer"
+
+	certificateDuration = 87600
+
+	composerCertificateName = "composer-cert"
+
+	composerComposerAPIServiceName = "osbuild-composer"
+	composerComposerAPIPortName    = "composer-api"
+	composerWorkerAPIServiceName   = "osbuild-worker"
+	composerWorkerAPIPortName      = "worker-api"
+
+	certificateSecretCACrtKey = "ca.crt"
+	certificateSecretCrtKey   = "tls.crt"
+	certificateSecretKeyKey   = "tls.key"
+
+	composerConfigMapName      = "osbuild-composer-config"
+	composerConfigMapKey       = "osbuild-composer.toml"
+	composerConfigTemplateFile = "osbuild-composer.toml"
+
+	composerProxyConfigMapName      = "osbuild-composer-proxy-config"
+	composerProxyConfigMapKey       = "envoy.yaml"
+	composerProxyConfigTemplateFile = "composer-proxy-config.yaml"
+
+	pgSSLModeDefault = "prefer"
+
+	composerDeploymentName         = "composer"
+	composerDeploymentTemplateFile = "composer-deployment.yaml"
 )
 
+// composerDeploymentParameters includes all the parameters needed to render the Composer Proxy Config and Deployment
+type composerDeploymentParameters struct {
+	ComposerDeploymentNamespace      string
+	ComposerDeploymentName           string
+	ComposerImageName                string
+	ComposerImageTag                 string
+	ProxyImageName                   string
+	ProxyImageTag                    string
+	ComposerAPIInternalPort          int
+	ComposerAPIExternalPort          int
+	WorkerAPIInternalPort            int
+	WorkerAPIExternalPort            int
+	PgSSLMode                        string
+	ProxyCertsDir                    string
+	PGSQLSecretName                  string
+	ComposerConfigMapName            string
+	ComposerCertsSecretName          string
+	ComposerCertsSecretPublicCertKey string
+	ComposerCertsSecretPrivateKeyKey string
+	ComposerCertsSecretCACertKey     string
+	ProxyConfigMapName               string
+}
+
 // OSBuildEnvConfigReconciler reconciles a OSBuildEnvConfig object
 type OSBuildEnvConfigReconciler struct {
 	client.Client
```

Comment:
Is the `Client` needed? All data management seems to be delegated to repositories.

Reply:
Good question. The `Client` is auto generated by the SDK for all controllers. So, I think we should consider removing it for all of them. But, I think that should be in a separate PR.
Should I remove it here?

Reply:
Please remove this one here and let's do the rest separately.